### ISSUE TITLE
WARNING: Don't upgrade to FBSDK v17 without supporting Limited Login (the Changelog is misleading)

### ISSUE THREAD

**Issue by ricsantos** [facebook/facebook-ios-sdk#2455]
### Checklist before submitting a bug report

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-ios-sdk/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](https://github.com/facebook/facebook-ios-sdk/blob/main/CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

### Xcode version

15.3

### Facebook iOS SDK version

17.0.0

### Dependency Manager

CocoaPods

### SDK Framework

Core

### Goals

Recently updated some iOS apps I work on to use FBSDK 17.0 so that the Privacy Manifests were included, as required by Apple.

As usual, I checked the changelog overview when updating, it reports:

https://github.com/facebook/facebook-ios-sdk/blob/main/CHANGELOG.md#1700

> ## 17.0.0
> ### Added
> * Added privacy manifests
>
> [2024-02-16](https://github.com/facebook/facebook-ios-sdk/releases/tag/v17.0.0) | [Full Changelog]

Which seems ok.

### Expected results

I expected the SDK to still work.

I expect the changelog overview to highlight any breaking changes, eg:
* Implement Limited Login or else the access token will no longer work when authenticating a user if they have denied ATT permission

### Actual results

Noticed some users were failing to authenticate on our servers (we send the access token to our server then use `npm passport-facebook-token` to authenticate and login or register the user). We started seeing these errors:

```
{
   "error":{
      "message":"Invalid OAuth access token - Cannot parse access token",
      "type":"OAuthException",
      "code":190,
      "fbtrace_id":"AUuBS..."
   }
})
```

Took a while to realise that, in FBSDK 17 and higher, if a user denies App Tracking (or later turns it off), the access token returned by FBSDKs `LoginManager.logIn(permissions` API will failed to be parsed.

### Steps to reproduce

1. Show user ATT prompt
2. Deny Tracking
3. Try to login using access token returned by `LoginManager.logIn(permissions`

### Code samples & details

```swift
Please update the changelog if that is possible, so others don't make the same mistake. 


Can't believe that such a breaking change was so poorly communicated, and the error message so vague.


Related v17 issues that show this was a surprise and the migration path not well communicated:

https://github.com/facebook/facebook-ios-sdk/issues/2365

https://github.com/facebook/facebook-ios-sdk/issues/2416

https://github.com/facebook/facebook-ios-sdk/issues/2417

https://github.com/facebook/facebook-ios-sdk/issues/2426

https://github.com/facebook/facebook-ios-sdk/issues/2431

https://github.com/facebook/facebook-ios-sdk/issues/2442

https://github.com/facebook/facebook-ios-sdk/issues/2446

https://github.com/facebook/facebook-ios-sdk/issues/2449

https://github.com/facebook/facebook-ios-sdk/issues/2451

and more...
```

**Comment by ricsantos** [comment]
Also found this issue which mirrors my sentiment:
https://github.com/facebook/facebook-ios-sdk/issues/2384

**Comment by ricsantos** [comment]
https://github.com/facebook/facebook-ios-sdk/issues/2365#issuecomment-2085542379

**Comment by krunalgoswamimoviepass** [comment]
Hey @ricsantos, We saw there is a updated version 17.0.2 available with some solution. Did you verified ? kindly help us out.

**Comment by ricsantos** [comment]
Yes 17.0.2 still needs Limited Login, any version >= 17 needs it.

**Comment by chaturj** [comment]
Hey @ricsantos,  i have some issue with multiple login limitation with ios 17 version with swift package menager version 17.0.2 please check and solve the issue asap.

**Comment by ricsantos** [comment]
@chaturj don't we all 😅

What's your problem?

**Comment by chaturj** [comment]
Hey @ricsantos,  i have issue with not open facebook app login with facebook for sdk version 17.0.2. it's work for sdk version 14.1.0.please see this screenshot for login with facebook open webpage 
![sdkversion17 0 2](https://github.com/user-attachments/assets/770aee56-7fc1-41db-8366-184d74acecac)
![sdkversion14 1 0](https://github.com/user-attachments/assets/a91c9768-6e0e-4d8a-89f1-0f1a33b18577)

**Comment by ricsantos** [comment]
Just use the web based login.

**Comment by gufoscuro** [comment]
@ricsantos I am using Capacitor in my hybrid app and bumping Capacitor to `v6.x.x`, and along with that, FBSDK is also bumped to v17 – from then on, the backend started failing to authenticate (so I guess that's the same issue you are reporting above)

I am using the limited login though – shouldn't that just work, even with sdk17, or I'm just misunderstanding the issue? I'd really appreciate if you can help me 🙏 thank you!

**Comment by FanYouXin** [comment]
16.3.1 works fine. If your face the problem : ITMS-91061: Missing privacy manifest ......
Can use this script to add privacyfill to your thirdparity framework.
https://github.com/crasowas/app_privacy_manifest_fixer

everybody can try to buy some transition time by downgrading the SDK to 16.3.1 and waiting for the server to support Limited Login.

**Comment by MirkoSrsens** [comment]
You can call. (These are Unity calls but you can google your specific case/platform)
                ATTrackingStatusBinding.RequestAuthorizationTracking(); 
Then you can check ATTrackingStatusBinding.GetAuthorizationTrackingStatus() if user denied it and give him pop up telling to go enable tracking in Privacy & Security > Tracking > AppName . This will force old token.

Worked for me since IOS store now requires PrivacyPolicy that is included in frameworks from 17+ versions, but that also comes with limited token disaster.
